src/v3/src/hooks/useOnSubmit.ts

```diff
@@ -262,9 +252,14 @@ export const useOnSubmit = (): (options: OnSubmitHandlerOptions) => Promise<void
         // No need to continue since onSuccess is called in registration hook callback
         return;
       }
+
+      if (!isClientTransaction) {
+        await executeWidgetHooks('before', hooksRef.current, newTransaction);
```

Comment:
does gen 2 have any situations where they would execute the `before` hook before a client transaction?

Reply:
cc: @glenfannin-okta

Reply:
I use `isClientTransaction` as an indicator that request did not succeed. 
In this case "before" and "after" hooks should not be executed for "terminal" form. As it's done for Gen2.

See test `should send "afterError" event (and not execute hooks for "terminal" form) if submit results with error`: https://github.com/okta/okta-signin-widget/pull/3376/files#diff-189986d3f484c4c778c28206906c6fb9e91e4d71404d94ad174eddba3e0acfdcR565